docs/source/development/config.rst

```diff
@@ -55,6 +55,17 @@ under :file:`profile_default`. If you want the default config files for the
 :mod:`IPython.parallel` applications, add ``--parallel`` to the end of the
 command-line args.
 
+.. note::
+
+    By default IPython will ignore any invalid configuration files, or any
+    unknown configuration option.
+
+.. versionadded:: 5.0
+
+    IPython can be configured to abort in case of invalid configuration file.
```

Comment:
I think we should mention that this doesn't catch misnamed keys or invalid values: 'invalid' means we couldn't load it at all.